onnx/fixmodel.py

```diff
@@ -0,0 +1,150 @@
+"""onnx model fixer
+
+This utility will fix up common model errors.
+
+1. Add a domain to the model object if there isn't one.
+2. Fix names of nodes, graphs, values, etc. to conform to the standard.
+
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import os
+import onnx
+from onnx import defs
+from onnx.defs import OpSchema
+import onnx.shape_inference
+
+import argparse
+
+def isempty(str):
+    return str == None or str == ''
+
+def replaceInvalidChars(name):
+
+    global modified
+
+    if name == '':
+        return name
+
+    n = name
+
+    if name[0].isnumeric():
+        n = "_" + n
+    n = n.replace("/","_").replace("-","_")
+
+    if n != name:
+        modified = True
+
+    return n
```

Comment:
But we can't be sure that the modified name is unique in the name-space it belongs to, right?

Reply:
That's correct, but this utility only fixes the identifier syntax issue, not duplication. It is possible that it introduces name collisions, since it replaces both '/' and '-' with a character that may actually be used in some other name, which is why this is just a utility, not a fool-proof tool. From what I've seen in models, this is good enough, but the real solution is to fix the problem at the source.